api/capi/src/nntrainer.cpp

```diff
@@ -388,7 +433,9 @@ int ml_train_model_get_layer(ml_train_model_h model, const char *layer_name,
                              ml_train_layer_h *layer) {
   int status = ML_ERROR_NONE;
   ml_train_model *nnmodel;
-  ML_NNTRAINER_CHECK_MODEL_VALIDATION(nnmodel, model);
+
+  ML_TRAIN_GET_VALID_MODEL_LOCKED(nnmodel, model);
+  ML_TRAIN_ADOPT_LOCK(nnmodel, model_lock);
```

Comment:
Do we need to set the lock when user just get the layer? May be lock when set the layer value or change value of layer?

Reply:
As getting the layer involves accessing the model elements, and user can delete the model, this lock is needed.
Later, we can move to reader/writer locks for better performance.